LPS-134354 Create the translation selector component

julien opened the pull request:
As specified by the [Lexicon documentation](https://docs.google.com/document/d/1wTEypASdVIKPTYdB0KygMwbLsCynBmpk94XkYyk357M/), we're adding the
translation selector component. This is the first version of the React
component, implemented with Clay components and TypeScript.

The translation selector has 3 variants.

Here's, the default variant, which allows managing translations:
![01-translation-selector-default](https://user-images.githubusercontent.com/5572/122921335-a5fcdc80-d362-11eb-8174-3ff1f0e5f9b3.png)
(*Note that the red border is only part of the screenshot, not the actual component*)

It can also be used as a "language selector"
![02-language-selector](https://user-images.githubusercontent.com/5572/122921397-b8771600-d362-11eb-97d3-6ef487e67a2f.png)

Or as a "translation selector"
![03-translation-selector](https://user-images.githubusercontent.com/5572/122921455-cc227c80-d362-11eb-90cd-826b335dcdb3.png)

The translation selector's button can also be rendered as "small"
![04-button-size-small](https://user-images.githubusercontent.com/5572/122921531-e197a680-d362-11eb-9126-69e6b61872b3.png)

liferay-continuous-integration commented:
Please only forward critical changes to Brian Chan during stabilization. Nonurgent changes should wait until the ongoing 7.4 DXP EP2 & CE GA2 release has been completed. For more details on the release timeline and status, see <a href="https://liferay.slack.com/archives/C03BTCQBQ">product-delivery</a>.

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-25.liferay.com/job/test-portal-source-format/1613/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-25.liferay.com//userContent/jobs/test-portal-source-format/builds/1613/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:sf - 0 out of 1 jobs passed in 3 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/3a8ab55f0466afd7405223636a5cef15307a31dd">3a8ab55f0466afd7405223636a5cef15307a31dd</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/julien/liferay-portal/tree/LPS-134354">LPS-134354</a><br/>Branch GIT ID: <a href="https://github.com/julien/liferay-portal/commit/72da35690c8137c8fe17ae4e1d8b9f6061c0e702">72da35690c8137c8fe17ae4e1d8b9f6061c0e702</a></p>0 out of 1jobs PASSED<div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-25.liferay.com/job/test-portal-source-format/1613/"><strike><strong>test-portal-source-format</strong></strike></a></li></ul></div><h5>For more details click <a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-source-format/builds/1613/jenkins-report.html">here</a>.</h5><pre><code>     [java] java.lang.Exception: Found 4 formatting issues:
     [java] 1: ./modules/apps/frontend-js/frontend-js-translation-web/package.json expected:&lt;...ayui/drop-down": "3.[30].0",
     [java] 		"@clayui/form...&gt; but was:&lt;...ayui/drop-down": "3.[29].0",
     [java] 		"@clayui/form...&gt;
     [java] 2: ./modules/apps/frontend-js/frontend-js-translation-web/bnd.bnd expected:&lt;...d-js-translation-web[]&gt; but was:&lt;...d-js-translation-web[
     [java] ]&gt;
     [java] 3: ./modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/content/Language.properties expected:&lt;[manage-translations=Manage Translations
     [java] untranslated=Not Translated]&gt; but was:&lt;[default=Default
     [java] manage-translations=Manage Translations
     [java] translated=Translated
     [java] untranslated=Not Translated
     [java] ]&gt;
     [java] 4: ./modules/apps/frontend-js/frontend-js-translation-web/build.gradle expected:&lt;...tra:petra-string")
     [java] \}[]&gt; but was:&lt;...tra:petra-string")
     [java] \}[
     [java] ]&gt;
     [java] 
     [java] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:483)
     [java] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:312)
[stopwatch] [run.batch.test.action: 1:04.193 sec]
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:572: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/portal-impl/build.xml:706: Java returned: 1
      [get] Getting: http://test-1-25/job/test-portal-source-format/1613//consoleText
      [get] To: /opt/dev/projects/github/liferay-portal/20210622050603051.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/20210622050603051.txt
  [typedef] Could not load definitions from resource org/apache/maven/artifact/ant/antlib.xml. It could not be found.
  [taskdef] Could not load definitions from resource org/jacoco/ant/antlib.xml. It could not be found.
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/null1236394178.properties

merge-test-results:
    [mkdir] Created dir: /opt/dev/projects/github/liferay-portal/test-results
[junitreport] Processing /opt/dev/projects/github/liferay-portal/test-results/TESTS-TestSuites.xml to /tmp/null298949979
[junitreport] Loading stylesheet jar:file:/opt/java/ant/lib/ant-junit.jar!/org/apache/tools/ant/taskdefs/optional/junit/xsl/junit-frames.xsl</code></pre></details></html>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/1774/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-25.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/1774/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-25.liferay.com/job/test-portal-source-format/1613/">test-portal-source-format#1613</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-source-format/builds/1613/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1184">liferay-frontend#1184</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1541931822">[master] ci:test:sf - julien &gt; liferay-frontend - PR#1184 - 2021-06-22[05:03:33]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-25.liferay.com/job/publish-testray-report/4765/">publish-testray-report#4765</a><br/></div></div>

julien commented:
`gradlew formatSource`,`yarn checkFormat`, and `ant format-source-current-branch` (from the `portal-impl`) directory all pass locally. Any ideas @liferay-frontend?

julien commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-22.liferay.com/job/test-portal-source-format/6660/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-22.liferay.com//userContent/jobs/test-portal-source-format/builds/6660/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/3a8ab55f0466afd7405223636a5cef15307a31dd">3a8ab55f0466afd7405223636a5cef15307a31dd</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/julien/liferay-portal/tree/LPS-134354">LPS-134354</a><br/>Branch GIT ID: <a href="https://github.com/julien/liferay-portal/commit/df27174affc5235e22fc730c34c16ac3525a4881">df27174affc5235e22fc730c34c16ac3525a4881</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-22.liferay.com/job/test-portal-source-format/6660/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-22.liferay.com/userContent/jobs/test-portal-source-format/builds/6660/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-22.liferay.com/job/test-portal-source-format/6660/">test-portal-source-format#6660</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-22.liferay.com/userContent/jobs/test-portal-source-format/builds/6660/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1184">liferay-frontend#1184</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1541937386">[master] ci:test:sf - julien &gt; liferay-frontend - PR#1184 - 2021-06-22[05:20:51]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-22.liferay.com/job/publish-testray-report/1079/">publish-testray-report#1079</a><br/></div></div>

jonmak08 commented:
@julien, Looks like all the tests have passed now.

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 10 out of 10 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 23 out of 23 jobs passed in 2 hours 60 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/3a8ab55f0466afd7405223636a5cef15307a31dd">3a8ab55f0466afd7405223636a5cef15307a31dd</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/3a8ab55f0466afd7405223636a5cef15307a31dd">3a8ab55f0466afd7405223636a5cef15307a31dd</a><br/>Jenkins Build URL: <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)//2027">Acceptance Upstream DXP (master) #2027</a></p><details><summary><strong>ci:test:stable - 10 out of 10 jobs PASSED</strong></summary><details><summary><strong>10 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12944/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13148/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13149/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13147/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474998/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474996/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474878/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/330026/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13146/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474997/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 23 out of 23 jobs PASSED</strong></summary><details><summary><strong>23 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/1774/">test-portal-acceptance-pullrequest(master)</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/330027/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8/0</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/417084/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12944/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12947/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/417085/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13148/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13149/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8/0</a></li><li><a href="https://test-1-18.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/229919/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8/0</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/389907/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13147/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/466147/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474998/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12946/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474877/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474996/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/12945/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474878/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/330026/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/389906/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13146/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/13145/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8/0</a></li><li><a href="https://test-1-11.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/474997/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><h5>For more details click <a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/1774/jenkins-report.html">here</a>.</h5></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/1774/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/1774/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-25/jobs/test-portal-acceptance-pullrequest(master)/builds/1774/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest(master)/1774/">test-portal-acceptance-pullrequest(master)#1774</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-25.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/1774/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/1184">liferay-frontend#1184</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1542008914">[master] ci:test:relevant - julien &gt; liferay-frontend - PR#1184 - 2021-06-22[05:03:29]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-25.liferay.com/job/publish-testray-report/4785/">publish-testray-report#4785</a><br/></div></div>

diegonvs commented on modules/apps/frontend-js/frontend-js-translation-web/types/src/main/resources/META-INF/resources/TranslationSelector.d.ts:
```diff
@@ -0,0 +1,81 @@
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap?: string;
```
We don't have the habit to receive `spritemap` property in DXP React component due to ClayIconSpriteContext. I think you could remove it.

julien replied:
Yeah we could that's true, I'm just wrapping this in a context anyway so I'll do that: +1

julien replied:
Sorry had to push force because I rebased with `master`, but that's updated in [`5673627`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/567362728b147374d72fe070f7628483ba917086)

diegonvs commented on modules/apps/frontend-js/frontend-js-translation-web/types/src/main/resources/META-INF/resources/index.d.ts:
```diff
@@ -0,0 +1,23 @@
+ */
+
+import {IProps} from './TranslationSelector';
+declare global {
```
Did you generated the `.d.ts` file and add the declaration of Language.get there?

If yes, I think We can declare Language.get in the component file(`.tsx`) which is using and just generate `.d.ts`.

julien replied:
> Did you generated the .d.ts file and add the declaration of Language.get there?

As far as I can tell generating the `d.ts` file is done automagically, and yes; `Language.get` is  declared here 

> If yes, I think We can declare Language.get in the component file(.tsx) which is using and just generate .d.ts.

All components in this (OSGi) module that use strings will end up using `Liferay.Language.get` because we need i18n, and we're actually having a discussion about those globals [here](https://liferay.slack.com/archives/C3JBR21HA/p1624358954210300) on Slack, so let me know

diegonvs commented on modules/apps/frontend-js/frontend-js-translation-web/test/TranslationSelector.js:
```diff
@@ -0,0 +1,107 @@
+		);
+
+		expect(container).toMatchSnapshot();
+	});
```
You can add a test for testing `onSelectedLocaleChange` callback :)

julien replied:
Yes, why not

julien replied:
Updated in [`2752f20`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/2752f205b152e876ce51c3898d5241a89d802b89)

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx:
```diff
@@ -0,0 +1,221 @@
+
+export type ButtonSize = null | 'regular' | 'small';
+
+export type Variant =
+	| null
+	| 'default'
+	| 'language selector'
+	| 'translation selector';
```
Perhaps in this case it would be better to define shorter names for variation, for example: `type Variant = 'language' | 'translation';`, the `default` can always be when the property value is `undefined`. What do you think?

julien replied:
I think it's a good proposal, going to update

julien replied:
Updated in [`09a16fd`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/09a16fd28de164ce4eb3ebe6162531ca0c30cd91)

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/index.tsx:
```diff
@@ -0,0 +1,30 @@
+	};
+}
+
+export default function (container: Element, props: IProps) {
+	render(<TranslationSelector {...props} />, {}, container);
+}
```
Is the idea of directly calling `render` because we want to allow this component to also be used outside of taglib like `react:component`?

diegonvs replied:
When reading this code it brought this thing too but I forgot to make the question :p

julien replied:
In dxp, we can't use `ReactDOM.render` directly (that throws an `eslint` error) , and yes we also want this component to be used directly with no taglib.

matuzalemsteles replied:
Thanks @julien for info

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx:
```diff
@@ -0,0 +1,221 @@
+}: IProps) => {
+	const [active, setActive] = React.useState(false);
+
+	const defaultLocale = locales[0];
```
Maybe it wouldn't be better to leave this configurable? in this case we are always relying on the default value to always be first in the list. And in this case, whoever passes the data must always organize list before rendering.

I think this is also a criterion for Translation, right? always list the default language as first?

julien replied:
Yes sure, this was me being lazy, we can totally make it configurable - let me fix that.

julien replied:
Yes you're right, the default language should be the first one we "render" in that list.

julien replied:
Updated in [`84fb294`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/84fb294ca0bf424a10c89db94cfa4a2fab0b00f5)

matuzalemsteles replied:
Thanks for checking @julien

matuzalemsteles reviewed (commented):
Well, a great job, overall it looks good, I just added some API details and some behaviors to clarify doubts.

julien commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx:
```diff
@@ -0,0 +1,221 @@
+}: IProps) => {
+	const [active, setActive] = React.useState(false);
+
+	const defaultLocale = locales[0];
+
+	if (!selectedLocale) {
+		selectedLocale = defaultLocale;
+	}
```
Sure

julien replied:
Updated in [`a2f9dcc`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/a2f9dcc12af5629f48e9a0feb70bbddf2586d4f6)

julien commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx:
```diff
@@ -0,0 +1,221 @@
+		manageTranslations: string;
+	};
+
+	/**
+	 * Button size to use "regular" or "default"
+	 */
+	buttonSize?: ButtonSize;
```
Makes sense, I'll go with `small`. I'm always in favor of adding less code possible.

julien replied:
Updated in [`6fbcb55`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/6fbcb550bb3264a184ad096f0d036fb50eed47e8)

julien commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx:
```diff
@@ -0,0 +1,208 @@
+	/**
+	 * Callback that gets called the "Manage Translations" drop down item is clicked
+	 */
+	onManageTranslations?: () => void;
```
@matuzalemsteles + @diegonvs I'm probably going to remove this because we are always going to do the same thing: open the translation modal (which I'm working on right now) cc @dsanz

dsanz replied:
The presence of that behavior (i.e. providing a "manage translation" button and opening the modal) depends on the TranslationSelector being in **admin mode** or not ([explanation](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/1593639193/LPS-133329+New+Translation+Manager#Translation-selector.-One-component%2C-multiple-roles.)).

As the translation admin is a separate component which is meant to be used on its own (and not only from the translation selector), I'd say that what translation selector could do is to compose the translation admin when invoked in admin mode, and perhaps register some listener on it so that when the user confirms the available languages, all instances of translation selector (both react and jsp-based) can update the list of available locales.

However that's an impl detail which basically means I'm not sure what I'm talking about
![image](https://user-images.githubusercontent.com/861014/123104857-98fded00-d437-11eb-9325-6ce43b55c6c7.png)

matuzalemsteles replied:
Oh makes sense, well when I saw this I figured it would be done outside the TranslationSelector, but yes, it makes sense to remove it and add the internal implementation to the component since we have a variant for that.

julien replied:
> As the translation admin is a separate component which is meant to be used on its own (and not only from the translation selector)

Thanks @dsanz that already clears some doubts: we will need the "modal" to be a separate component so that it can be used on its own, but I still think that this `onManageTranslations` callback is not needed, @matuzalemsteles let me know if that also makes sense to you when you get a chance.

julien commented:
Converted this to draft because I'll be adding the "Translation Modal" on top of this.

julien commented:
@matuzalemsteles + @diegonvs I've added the `TranslationModal` component and also switched from TypeScript to (good old) JavaScript, reason is that we'll be using this component in older components and we don't have a way of loading (i.e. `require`ing) TS code from there.

diegonvs commented:
> @matuzalemsteles + @diegonvs I've added the TranslationModal component and also switched from TypeScript to (good old) JavaScript, reason is that we'll be using this component in older components and we don't have a way of loading (i.e. requireing) TS code from there.

This is weird because We're providing `frontend-js-react-web` which is totally written using TS and people are having any errors.

> component in older components and we don't have a way of loading (i.e. requireing) TS code from there.

In this part, are you meaning requiring `.ts` files on `aui:script` for example? Could you explain more about it?

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+			</ClayModal.Header>
+
+			<ClayModal.Body scrollable>
+				<ClayManagementToolbar aria-label="Management toolbar">
```
This should come from `ariaLabels ` prop

julien replied:
You're right, I'll fix that

julien replied:
Updated in [`83cf3fb`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/83cf3fbb4cb744c21ab314986d7059b4a87e9046)

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+							<ClayInput.GroupItem>
+								<ClayInput
+									aria-label={Liferay.Language.get('search')}
+									className="form-control input-group-inset input-group-inset-after"
```
I believe we provide `insetAfter` as a prop on ClayInput

julien replied:
Ah Ok, I'll update that (still getting around the API of some components I haven't used much)

julien replied:
Updated in [`ab35f14`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/ab35f14ab8b71a14ee8cb3edc9873ae8b469fe82)

bryceosterhaus replied:
> still getting around the API of some components I haven't used much

no worries! I still have to check the docs 75% because I forget what APIs we offer, even though I wrote a lot of them 😂 

Pretty much any time I find myself having to add a clay class to a clay component, I realize we either already offer the API or we need to add it in. Main exception is for simple util classes

julien replied:
> no worries! I still have to check the docs 75% because I forget what APIs we offer, even though I wrote a lot of them :joy: 

I can imagine, there are a bunch of components in there :smile:

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+									aria-label={Liferay.Language.get('search')}
+									className="form-control input-group-inset input-group-inset-after"
+									onInput={() => {
+										const value = inputRef.current?.value;
```
How come you went with a ref instead of using the event.target?

```jsx
onInput={(event) => {
    const {value} = event.target;
    //...
}}

julien replied:
That's a good question ... I actually used a ref, because I thought that was the React way to do things, but using event.target did come to mind. I'll use that next time, thanks

julien replied:
Updated in [`8a9dfad`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/8a9dfad58eeeec19dfd11eed9e325170680b3266) - sorry this might be the "hardest" thing to review, but it does look nicer

bryceosterhaus replied:
Typically the "react way"™️  is to use [Controlled Components](https://reactjs.org/docs/forms.html#controlled-components). Although, there are cases where controlled components cause issues, so I have actually seen people use refs to get around those issues.

More thoughts on controlling inputs... Additionally, controlling an input isn't _always_ necessary. You really only need to control it when you want to do something based on that state. For example, if you just have a simple form you can leave all the inputs uncontrolled and just get the data on form submit. However, if you want to render something based on the value of one of those inputs then you need to control it. See below for an example or [codesandbox](https://codesandbox.io/s/trusting-mendel-t4uzh?file=/src/App.js).

```jsx
const UncontrolledComponent = () => {
  function handleSubmit(event) {
    event.preventDefault();

    const data = {
      firstName: event.target.firstName.value,
      lastName: event.target.lastName.value,
      age: event.target.age.value
    };

    // POST your data here
  }

  return (
    <form onSubmit={handleSubmit}>
      <input name="firstName" />
      <input name="lastName" />
      <input name="age" type="number" />
      <button>Submit</button>
    </form>
  );
};

// Note below how we want to update the submit button based on the value of the inputs.

const ControlledComponent = () => {
  const [firstName, setFirstName] = React.useState("");
  const [lastName, setLastName] = React.useState("");
  const [age, setAge] = React.useState("");

  function handleSubmit(event) {
    event.preventDefault();

    const data = {
      firstName,
      lastName,
      age
    };

    // POST your data here
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        name="firstName"
        onChange={(event) => setFirstName(event.target.value)}
        value={firstName}
      />
      <input
        name="lastName"
        onChange={(event) => setLastName(event.target.value)}
        value={lastName}
      />
      <input
        name="age"
        onChange={(event) => setAge(event.target.value)}
        type="number"
        value={age}
      />
      <button disabled={!firstName.length || !lastName.length || !age.length}>
        Submit
      </button>
    </form>
  );
};
```

So for your specific case, it is helpful to control the input because you are filtering based on the input.

julien replied:
Thanks, once again that's a very helpful explanation and code example. Makes much more sense now.

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+									}}
+									placeholder={Liferay.Language.get('search')}
+									ref={inputRef}
+									type="text"
```
I believe this is already the default "type" on the component, so there is no need to specify it (although I guess it doesn't necessarily hurt to leave it).

julien replied:
You're right, if it's not needed it doesn't need to be there.

julien replied:
Done in [`a0b4d93`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/a0b4d93f9e32601df6cca2be04b9977a309b6c87)

bryceosterhaus reviewed (commented):
Looks good! Just had a few minor comments

I was curious though, what was the thinking through making a new module for this component? I wonder if we might create a more generic module to store an assortment of react components used across teams in DXP. Otherwise, we grow the number of osgi modules linearly with the amount of shared react components we have.

I guess the other place this might live would be the `frontend-taglib-clay` module.

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [icon, setIcon] = React.useState('search');
+	const inputRef = React.createRef();
```
Why keep the input uncontrolled? we could just keep the component controlled instead of using a reference to change the component value in some cases, it is usually preferable if you use the controlled component strategy, so you are free to make strategies like the previous comment.

https://reactjs.org/docs/uncontrolled-components.html

julien replied:
I agree with both what @bryceosterhaus and you said, there's no need for the input to be uncontrolled, my mistake

julien replied:
Updated in [`8a9dfad`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/8a9dfad58eeeec19dfd11eed9e325170680b3266) - sorry this might be the "hardest" thing to review, but it does look nicer

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+											return;
+										}
+
+										const result = Object.values(
+											visibleLocales
+										).filter((locale) => {
+											return locale.label.match(
+												new RegExp(value, 'i')
+											);
+										});
```
There are some complicated cases of dealing with filters this way, when you use the same information source and update it filtered, as this is a controlled filter with a limited list, the best world for this is to keep the filter being done in time rendering anyway this is being done too but you can handle the filter being done inside a `useMemo` with a dependency to a `value`, this is quite common to see in some like this and a little more reactive to changing the value.

For example, the value of `visibleLocales` will be reset to the original value which is `locales` only if the user clicks on the icon, this usually doesn't cover all interactions, the user can just clear all entered values and the behavior default of listing all values are not listed, as you type the filter it will be based on the most recent `visibleLocales`, so if the user gives up on what he just typed and then writes a new value, the chance of not showing the correct result is very high, in this case, the original state should always be considered instead of the recent state of the filter.

Controlling all these cases makes it difficult when you need to filter and store a state, so I have a preference and much more common in other applications and cases like autocomplete are reactive to `value` at render time.

```jsx
const filteredLocales = useMemo(() => {
  return Object.values(
    locales
  ).filter((locale) => locale.label.match(
    new RegExp(value, 'i')
  ))
}, [locales, value]);
```

I'm not sure but `locales` already looks like an `Array` so it doesn't need an `Object.values(locales)`.

julien replied:
Thanks for the explanation @matuzalemsteles. I'll use the `useMemo` hook, and check if  `Object.values` is really needed here

julien replied:
Updated in [`8a9dfad`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/8a9dfad58eeeec19dfd11eed9e325170680b3266) - sorry this might be the "hardest" thing to review, but it does look nicer

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+					</ClayManagementToolbar.Search>
+
+					<ClayManagementToolbar.ItemList>
+						<ClayDropDown
+							active={active}
+							onActiveChange={setActive}
+							trigger={<ClayButtonWithIcon symbol="plus" />}
+						></ClayDropDown>
```
Nothing too critical here, but it's more common to use more self-closing tag in JSX.

julien replied:
True, I'll do that.

julien replied:
Updated in [`9349512`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/93495121c4028424aee39bc9746b60aad2597387)

matuzalemsteles commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+							<ClayTable.Cell headingCell>
+								{Liferay.Language.get('status')}
+							</ClayTable.Cell>
+							<ClayTable.Cell headingCell></ClayTable.Cell>
```
The same here
```suggestion
							<ClayTable.Cell headingCell />
```

julien replied:
Yes, good point

julien replied:
Updated in [`9349512`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/93495121c4028424aee39bc9746b60aad2597387)

matuzalemsteles reviewed (commented):
Well, it looks good, I added some comments about some behaviors that can make a bug in the filter part and other small comments.

I think I agree with Bryce on this, I think maybe the best place for this should be the `frontend-js-react-web`, because we're already doing this with hooks but I understand that this module should be used elsewhere not React or just use as `react:component` and it looks like we have some limitations with TypeScript as well.

julien commented:
> I was curious though, what was the thinking through making a new module for this component? I wonder if we might create a more generic module to store an assortment of react components used across teams in DXP. Otherwise, we grow the number of osgi modules linearly with the amount of shared react components we have.

> I guess the other place this might live would be the frontend-taglib-clay module.

Hey @bryceosterhaus, thanks for the review. To be honest, I did think about it but I didn't have a strong opinion about it, having this in `frontend-taglib-clay` is probably a better option, I'll move those there, I also think adding more OSGi module isn't a good idea.

julien commented on modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js:
```diff
@@ -0,0 +1,201 @@
+	translations,
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [icon, setIcon] = React.useState('search');
```
That's right, there's no need for the state here, this is more simple and does the job.

julien commented:
@bryceosterhaus, @matuzalemsteles thanks again for the help and advice on this review: I think your comments brought a lot of improvements. I finally moved everything to `frontend-taglib-clay`, which might make the review process a bit difficult, but each commit should be "clear" enough. Let me know what you think when you can (keeping in mind there will be more work once I integrate the Java part).

bryceosterhaus commented on modules/apps/frontend-taglib/frontend-taglib-clay/package.json:
```diff
@@ -82,11 +82,13 @@
 		"react-dom": "16.12.0"
 	},
 	"name": "frontend-taglib-clay",
+	"main": "index.js",
```
I'm curious about this change, was this required for something specific? Or was this primarily because you added the index.js file?

julien replied:
Good question, and this is still a bit of a "draft": the goal is to be able to import components that are exported from `index.js` in any other module. In my case here I'm just using it in `portal-web`, because I want to render the `TranslationSelector` from the [`InputLocalized`](https://github.com/liferay/liferay-portal/blob/167f47e46c2ea38101a1cfbf6d2ba237e138b484/portal-web/docroot/html/taglib/ui/input_localized/page.jsp) tag, but in the long run we'd also want for devs to be able to just render these "translation" components in their React app too.

bryceosterhaus commented on modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/index.js:
```diff
@@ -0,0 +1,25 @@
+
+export {TranslationModal};
+
+export default function (container, props) {
+	render(<TranslationSelector {...props} />, {}, container);
+}
```
I'm guessing you just copied this over from the original module so it may have just been an oversight, but it is odd to me that we are adding a main entry point to `frontend-taglib-clay` and exporting a function to render a very specific component. This would make sense if it was its own module, but I don't think we should do this in a large "clay" module like this one.

Do you know where this component will be used at specifically or how it is used (in a react component or taglib)? Will it be used across multiple modules and in different UIs? I think this all goes back to the "where does this component live?" question. I read up on the LPS and I can't quite put together where this is going to be implemented at. If we know where this will be implemented at, it will help give an idea of how to expose it.

julien replied:
Yes, you guessed right.

It is totally odd to be exporting one function that just renders a specific component - but I just copied the file from the `frontend-js-translation-web` module we decided to discard guessing we'd have a discussion about this.

The `TranslationSelector` and `TranslationModal` components could be used from any OSGi module that uses React ... but we also want to replace [this part](https://github.com/liferay/liferay-portal/blob/167f47e46c2ea38101a1cfbf6d2ba237e138b484/portal-web/docroot/html/taglib/ui/input_localized/page.jsp#L136-L222) of the  `InputLocalized` tag to render the `TranslationSelector` depending on an attribute (I haven't sent that yet) 

Maybe what I could do is export a "named" function instead of a default one that makes it more clear: for example `renderTranslationSelector`.

bryceosterhaus replied:
> Maybe what I could do is export a "named" function instead of a default one that makes it more clear: for example renderTranslationSelector.

Yeah I like this idea, but instead of exporting a function that renders it, could we just export the component itself? That way we can defer the rendering bit to whoever is consuming it. That way it can be used both in existing React code or anywhere else.

This file would then just be

```js
import TranslationModal from './translation/TranslationModal';
import TranslationSelector from './translation/TranslationSelector';

export {TranslationSelector, TranslationModal};

// I don't know if it will complain about no default export... If it does, you can add an empty object for now
export default {}
```

julien replied:
@bryceosterhaus that would be ideal. 
However in this case, I haven't found a way to define a function that renders the `TranslationSelector` in the `portal-web` module, so I defined a `renderTranslationSelector` function that I also export ... 

I'm thinking that maybe this is not the best solution: maybe the tag could load the required (JS) modules to make this happen, or maybe rendering the react component in that taglib doesn't make sense at all (after all, if this needs to be backported we could face issues). I'll speak with @dsanz about the issue just to see if he has ideas.

bryceosterhaus reviewed (commented):
The code looks great! I'm only concerned about where this will live and how to expose it to other teams. I think I'm still not quite clear on that, but that is likely due to my own lack of understanding. Once we can get an idea of how they want to implement it, it will help us know where it should live and how teams and consume it.

matuzalemsteles commented:
I took a look again seeing the recent changes and the component looks good to me, I miss the removal interactions but I imagine this will be the next step after solving some issues with the bundler.

julien commented:
@matuzalemsteles I added the missing logic to manage translations in [d6d4d9f](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/d6d4d9f0fda2d095ee0ddf9d8d59acb415402c0a)

diegonvs commented on modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/translation/TranslationModal.js:
```diff
@@ -88,12 +93,90 @@ const TranslationModal = ({
+										<ClayDropDown.Item
+											key={index}
+											onClick={() => {
+												var deletedLocaleIndex = deletedLocales.indexOf(
```
I think this `var` is not intended to be here. Also, What do you think in using a Set for deletedLocales? Since locales cannot be duplicated and it will save some LOCs to find and remove a given locale.

julien replied:
@diegonvs you're looking at an outdated commit/diff, `var` has been replaced by `const` here. 
As for using `Set` sure, I'll do that

julien replied:
Updated in [https://github.com/liferay-frontend/liferay-portal/commit/32d7ab2a9c6aea8787bffa20915918b37673b077](https://github.com/liferay-frontend/liferay-portal/commit/32d7ab2a9c6aea8787bffa20915918b37673b077)

matuzalemsteles commented on modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/translation/TranslationModal.js:
```diff
@@ -0,0 +1,264 @@
+											<ClayIcon
+												className="inline-item"
+												onClick={() => {
+													filteredLocales.splice(
+														index,
+														1
+													);
+
+													deletedLocales.add(locale);
+													setDeletedLocales(
+														deletedLocales
+													);
+
+													setDeletedLocale(locale);
+
+													setInitialLocales(
+														filteredLocales
+													);
```
I think this logic could be simplified a bit, for example `filteredLocales` doesn't need to be modified, because we just need to filter the `initialLocales` by removing the locale and `filteredLocales` will be updated on rendering.

```js
deletedLocales.add(locale);
setDeletedLocales(
  deletedLocales
);

setDeletedLocale(locale);

setInitialLocales(
  initialLocales.filter(({label}) => label !== locale.label)
);
```

It just avoids the side effect by directly changing the value of `filteredLocales` and it also avoids losing locales when the user is searching something and we use it to update the `initialLocales`.

matuzalemsteles commented on modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/translation/TranslationModal.js:
```diff
@@ -0,0 +1,264 @@
+			setDeletedLocales(deletedLocales);
+		}
+
+		filteredLocales.splice(locale.index, 0, locale);
+		setInitialLocales(filteredLocales);
+		setActive(false);
+		setDeletedLocale(null);
```
I think the same situation compared here above, we can avoid changing `filteredLocales` and also use it to update the `initialLocales`.

```js
setInitialLocales([...initialLocales, locale]);
setActive(false);
setDeletedLocale(null);
```

julien commented:
Closing this one because it's not finished yet and we're going to resend everything as part of [LPS-134356](https://issues.liferay.com/browse/LPS-134356)
